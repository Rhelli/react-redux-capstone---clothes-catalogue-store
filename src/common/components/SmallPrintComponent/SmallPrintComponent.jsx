import React from 'react';
import styles from './SmallPrintComponent.module.scss';

const SmallPrintComponent = () => (
  <footer className={styles.footer}>
    <p className={styles.disclaimer}>
      All images are property of ASOS.com. This site is not intended for any commercial purpose.
    </p>
    <p className={styles.name}>
      Made with love, hard work, sweat and tears. Rory Hellier 2021&copy;
    </p>
  </footer>
);

export default SmallPrintComponent;
