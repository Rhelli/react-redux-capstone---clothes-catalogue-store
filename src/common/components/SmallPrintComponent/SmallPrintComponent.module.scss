.footer {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 60px;
  background-color: #ede4dc;
  border-top: 1px solid #cec7c1;

  p {
    text-align: center;
  }

  .disclaimer {
    font: 400 14px/18px 'Source Sans Pro', monospace;
  }

  .name {
    font: 400 12px/20px 'Source Sans Pro', monospace;
  }
}
