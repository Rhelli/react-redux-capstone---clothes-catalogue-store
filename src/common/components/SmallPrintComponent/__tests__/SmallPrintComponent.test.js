import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import SmallPrintComponent from '../SmallPrintComponent';

describe('The appearance of the SmallPrintComponent footer', () => {
  it('Matches the SmallPrintComponent snapshot', () => {
    const SmallPrintSnapshot = renderer.create(<SmallPrintComponent />).toJSON();
    expect(SmallPrintSnapshot).toMatchSnapshot();
  });

  it('Contains two children in its entirety', () => {
    const { container } = render(<SmallPrintComponent />);
    expect(container.firstChild.childElementCount).toEqual(2);
  });

  it('Has two children, both paragraphs', () => {
    const { container } = render(<SmallPrintComponent />);
    expect(container.firstChild.firstChild.nodeName).toBe('P');
    expect(container.firstChild.lastChild.nodeName).toBe('P');
  });

  it('Contains a disclaimer in the first paragraph element', () => {
    const { getByText } = render(<SmallPrintComponent />);
    getByText('All images are property of ASOS.com. This site is not intended for any commercial purpose.');
  });

  it('Contains the website author and attributes as the second element', () => {
    const { getByText } = render(<SmallPrintComponent />);
    getByText('Made with love, hard work, sweat and tears. Rory Hellier 2021Â©');
  });
});
