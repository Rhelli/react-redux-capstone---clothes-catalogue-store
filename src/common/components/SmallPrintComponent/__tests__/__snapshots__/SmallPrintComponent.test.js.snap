// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance of the SmallPrintComponent footer Matches the SmallPrintComponent snapshot 1`] = `
<footer
  className="footer"
>
  <p
    className="disclaimer"
  >
    All images are property of ASOS.com. This site is not intended for any commercial purpose.
  </p>
  <p
    className="name"
  >
    Made with love, hard work, sweat and tears. Rory Hellier 2021Â©
  </p>
</footer>
`;
